libraries/dagster-iceberg/tests/io_managers/test_spark_io_manager.py

```diff
@@ -244,3 +286,41 @@ def test_iceberg_io_manager_with_hourly_partitioned_assets(
         "out_df = table.scan().to_arrow()",
         'assert out_df["partition"].to_pylist() == [datetime.datetime(2022, 1, 1, 3, 0), datetime.datetime(2022, 1, 1, 2, 0), datetime.datetime(2022, 1, 1, 1, 0)]',
     )
+
+
+def test_iceberg_io_manager_with_multipartitioned_assets(
+    asset_multi_partitioned_table_identifier: str,
+    catalog: Catalog,
+    io_manager: SparkIcebergIOManager,
+):
+    resource_defs = {"io_manager": io_manager}
+
+    for key in [
+        "a|2022-01-01",
+        "b|2022-01-01",
+        "c|2022-01-01",
+        "a|2022-01-02",
+        "b|2022-01-02",
+        "c|2022-01-02",
+    ]:
+        res = materialize(
+            [multi_partitioned],
+            partition_key=key,
+            resources=resource_defs,
+            run_config={
+                "ops": {"my_schema__multi_partitioned": {"config": {"value": "1"}}},
+            },
+        )
+        assert res.success
+
+    client = docker.from_env()
+    container = client.containers.get("pyiceberg-spark")
+
+    _exec_in_container(
+        container,
+        f'table = catalog.load_table("{asset_multi_partitioned_table_identifier}")',
+        "assert len(table.spec().fields) == 2",
+        'assert [f.name for f in table.spec().fields] == ["category_this", "date_this_day"]',
+        "out_df = table.scan().to_arrow()",
+        'assert out_df["date_this"].to_pylist() == [datetime.date(2022, 1, 2), datetime.date(2022, 1, 2), datetime.date(2022, 1, 2), datetime.date(2022, 1, 1), datetime.date(2022, 1, 1), datetime.date(2022, 1, 1)]',
```

Comment:
should probably check the other col too right?

Reply:
Oops, yeah. 🙈